File: phalcon/library/Export.php
<?php
namespace Library;

use Service\Base;

/* 导出 */
class Export extends Base {

  /* Excel内容 */
  static function Excel(array $data=[], array $param=[]): string {
    // 参数
    $param = array_merge([
      'borderColor'=>'#E2E4E8',      //边框颜色
      'titleColor'=> '#666',         //标题颜色
      'titleBgColor'=> '#F2F2F2',    //标题背景
    ],$param);
    // 内容
    $html = '<html>';
    $html .= '<style type="text/css">';
    $html .= 'table td{height: 32px; border: '.$param['borderColor'].' 1px solid;}';
    $html .= '.title{background-color: '.$param['titleBgColor'].'; color: '.$param['titleColor'].'; font-weight: bold;}';
    $html .= '</style>';
    $html .= '<table>';
    foreach($data as $k=>$v1){
      $html .= '<tr>';
      foreach($v1 as $v2){
        $html .= $k==0?'<td class="title">'.$v2.'</td>':'<td>'.$v2.'</td>';
      }
      $html .= '</tr>';
    }
    $html .= '</table>';
    $html .= '</html>';
    return $html;
  }

}

File: gin/library/export.go
package library

import (
	"webmis/util"
)

/* 导出 */
type Export struct{}

/* GExcel内容 */
func (e Export) Excel(data [][]interface{}, params map[string]interface{}) string {
	// 参数
	param := map[string]interface{}{
		"borderColor":  "#E2E4E8", //边框颜色
		"titleColor":   "#666",    //标题颜色
		"titleBgColor": "#F2F2F2", //标题背景
	}
	param = util.ArrayMerge(param, params)
	// 内容
	html := "<html>"
	html += "<style type=\"text/css\">"
	html += "table td{height: 32px; border: " + (&util.Type{}).Strval(param["borderColor"]) + " 1px solid;}"
	html += ".title{background-color: " + (&util.Type{}).Strval(param["titleBgColor"]) + "; color: " + (&util.Type{}).Strval(param["titleColor"]) + "; font-weight: bold;}"
	html += "</style>"
	html += "<table>"
	for k, v1 := range data {
		html += "<tr>"
		for _, v2 := range v1 {
			if k == 0 {
				html += "<td class=\"title\">" + (&util.Type{}).Strval(v2) + "</td>"
			} else {
				html += "<td>" + (&util.Type{}).Strval(v2) + "</td>"
			}
		}
		html += "</tr>"
	}
	html += "</table>"
	html += "</html>"
	return html
}
